drivers/video/video_common.c

```diff
@@ -83,3 +84,45 @@ void video_buffer_release(struct video_buffer *vbuf)
 		VIDEO_COMMON_FREE(block->data);
 	}
 }
+
+int video_get_range_u32(unsigned int cid, uint32_t *value, uint32_t min, uint32_t max, uint32_t def)
+{
+	switch (cid & VIDEO_CTRL_GET_MASK) {
+	case VIDEO_CTRL_GET_MIN:
+		*value = min;
+		return 0;
+	case VIDEO_CTRL_GET_MAX:
+		*value = max;
+		return 0;
+	case VIDEO_CTRL_GET_DEF:
+		*value = def;
+		return 0;
+	case VIDEO_CTRL_GET_CUR:
+		return 1;
+	default:
+		return -ENOTSUP;
+	}
+}
+
+int video_check_range_u32(const struct device *dev, unsigned int cid, uint32_t value)
+{
+	uint32_t min;
+	uint32_t max;
+	int ret;
+
+	ret = video_get_ctrl(dev, cid | VIDEO_CTRL_GET_MIN, &min);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = video_get_ctrl(dev, cid | VIDEO_CTRL_GET_MAX, &max);
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (value < min || value > max) {
```

Comment:
you should also check `min > max`

Reply:
I first added it as an `assert()`, but the MAX is not always fixed at build-time, i.e. could come from I2C read.
This can happen even when the software is correct => cannot be `assert()`!